src/tools/rbd/action/PersistentCache.cc

```diff
@@ -0,0 +1,159 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "tools/rbd/ArgumentTypes.h"
+#include "tools/rbd/Shell.h"
+#include "tools/rbd/Utils.h"
+#include "include/types.h"
+#include "include/rbd_types.h"
+#include "include/stringify.h"
+#include "common/errno.h"
+#include "common/Formatter.h"
+#include <iostream>
+#include <boost/program_options.hpp>
+
+namespace rbd {
+namespace action {
+namespace persistent_cache {
+
+namespace at = argument_types;
+namespace po = boost::program_options;
+
+namespace {
+
+const std::string PERSISTENT_CACHE_STATE = ".rbd_persistent_cache_state";
+
+} // anonymous namespace
+
+static int do_persistent_cache_invalidate(librbd::Image& image)
+{
+  int r;
+  uint64_t features;
+  r = image.features(&features);
+  if (r < 0) {
+    return r;
+  }
+
+  if (features & RBD_FEATURE_DIRTY_CACHE) {
```

Comment:
Why did you add the feature bit check here?

It looks like it matters in case the cache status metadata key is present but the feature bit is not set.  In that case with the old code we would have removed the key but with the new code it is left around.  Or am I misreading this?

Reply:
You're right. I should not make this change. In any case, the invalidate command should work.